src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/DataSetService.cs

```diff
@@ -180,4 +210,199 @@ private static DataSetVersionViewModel MapDataSetVersion(DataSetVersion dataSetV
+    {
+        dataSetVersion.FilterMetas = await _publicDataDbContext.FilterMetas
+            .AsNoTracking()
+            .Where(fm => fm.DataSetVersionId == dataSetVersion.Id)
+            .Include(fm => fm.Options)
+            .ThenInclude(fom => fom.MetaLinks)
+            .ToListAsync(cancellationToken: cancellationToken);
+
+        dataSetVersion.LocationMetas = await _publicDataDbContext.LocationMetas
+            .AsNoTracking()
+            .Where(lm => lm.DataSetVersionId == dataSetVersion.Id)
+            .Include(lm => lm.Options)
+            .ThenInclude(lom => lom.MetaLinks)
+            .ToListAsync(cancellationToken: cancellationToken);
+
+        dataSetVersion.IndicatorMetas = await _publicDataDbContext.IndicatorMetas
+            .Where(lm => lm.DataSetVersionId == dataSetVersion.Id)
+            .ToListAsync(cancellationToken: cancellationToken);
+
+        dataSetVersion.TimePeriodMetas = await _publicDataDbContext.TimePeriodMetas
+            .Where(lm => lm.DataSetVersionId == dataSetVersion.Id)
+            .ToListAsync(cancellationToken: cancellationToken);
+    }
+
+    private DataSetMetaViewModel MapVersionMeta(DataSetVersion dataSetVersion)
+    {
+        var filters = dataSetVersion.FilterMetas
+            .Select(MapFilterMeta)
+            .OrderBy(fm => fm.Label)
+            .ToList();
+
+        var indicators = dataSetVersion.IndicatorMetas
+            .Select(MapIndicatorMeta)
+            .OrderBy(im => im.Label)
+            .ToList();
+
+        var geographicLevels = dataSetVersion.LocationMetas
+            .Select(lm => lm.Level)
+            .ToList();
+
+        var locations = dataSetVersion.LocationMetas
+            .Select(MapLocationMeta)
+            .ToList();
+
+        var timePeriods = dataSetVersion.TimePeriodMetas
+            .Select(MapTimePeriod)
+            .OrderBy(tm => tm.Code.GetEnumValue())
+            .ThenBy(tm => tm.Period)
+            .ToList();
+
+        return new DataSetMetaViewModel
+        {
+            Filters = filters,
+            Indicators = indicators,
+            GeographicLevels = geographicLevels,
+            Locations = locations,
+            TimePeriods = timePeriods,
+        };
+    }
+
+    private static FilterMetaViewModel MapFilterMeta(FilterMeta filterMeta)
+    {
+        var options = filterMeta.Options
+            .Select(fom => MapFilterOptionMeta(filterMeta, fom))
+            .OrderBy(fom => fom.Label)
+            .ToList();
+
+        return new FilterMetaViewModel
+        {
+            Id = filterMeta.PublicId,
+            Hint = filterMeta.Hint,
+            Label = filterMeta.Label,
+            Options = options,
+        };
+    }
+
+    private static FilterOptionMetaViewModel MapFilterOptionMeta(
+        FilterMeta filterMeta,
+        FilterOptionMeta filterOptionMeta)
+    {
+        var publicId = filterOptionMeta.MetaLinks.Single(ml => ml.MetaId == filterMeta.Id).PublicId;
```

Comment:
This is inefficient as we have to search through all the links to find the public ID for each option, which is adding another nested loop.

Would suggest that we map using the `FilterOptionMetaLink` directly so that we can just do something like:

```cs
private static FilterOptionMetaViewModel MapFilterOptionMeta(
    FilterMeta filterMeta,
    FilterOptionMetaLink filterOptionMetaLink)
{
    return new FilterOptionMetaViewModel
    {
        Id = filterOptionMetaLink.PublicId,
        Label = filterOptionMetaLink.Option.Label,
        IsAggregate = filterOptionMetaLink.Option.IsAggregate,
    };
}
```